chainerrl/agents/dqn.py

```diff
@@ -246,73 +278,30 @@ def update(self, experiences, errors_out=None):
         loss.backward()
         self.optimizer.update()
 
-    def input_initial_batch_to_target_model(self, batch):
-        self.target_model(batch['state'])
-
     def update_from_episodes(self, episodes, errors_out=None):
-        has_weights = isinstance(episodes, tuple)
-        if has_weights:
-            episodes, weights = episodes
-            if errors_out is None:
-                errors_out = []
-        if errors_out is None:
-            errors_out_step = None
-        else:
-            del errors_out[:]
-            for _ in episodes:
-                errors_out.append(0.0)
-            errors_out_step = []
-
-        with state_reset(self.model), state_reset(self.target_model):
-            loss = 0
-            tmp = list(reversed(sorted(
-                enumerate(episodes), key=lambda x: len(x[1]))))
-            sorted_episodes = [elem[1] for elem in tmp]
-            indices = [elem[0] for elem in tmp]  # argsort
-            max_epi_len = len(sorted_episodes[0])
-            for i in range(max_epi_len):
-                transitions = []
-                weights_step = []
-                for ep, index in zip(sorted_episodes, indices):
-                    if len(ep) <= i:
-                        break
-                    transitions.append([ep[i]])
-                    if has_weights:
-                        weights_step.append(weights[index])
-                batch = batch_experiences(
-                    transitions,
-                    xp=self.xp,
-                    phi=self.phi,
-                    gamma=self.gamma,
-                    batch_states=self.batch_states)
-                assert len(batch['state']) == len(transitions)
-                if i == 0:
-                    self.input_initial_batch_to_target_model(batch)
-                if has_weights:
-                    batch['weights'] = self.xp.asarray(
-                        weights_step, dtype=self.xp.float32)
-                loss += self._compute_loss(batch,
-                                           errors_out=errors_out_step)
-                if errors_out is not None:
-                    for err, index in zip(errors_out_step, indices):
-                        errors_out[index] += err
-            loss /= max_epi_len
-
-            # Update stats
-            self.average_loss *= self.average_loss_decay
-            self.average_loss += \
-                (1 - self.average_loss_decay) * float(loss.array)
-
-            self.model.cleargrads()
-            loss.backward()
-            self.optimizer.update()
-        if has_weights:
-            self.replay_buffer.update_errors(errors_out)
+        assert errors_out is None
```

Comment:
Are we not supporting prioritized replay buffers? Maybe along with the assert, we can add a message is there is an assertion error?

Reply:
I added a message to it.